### PR TITLE
LPS-118704 Include the company as parameter to calculate the terms of an user

### PR DESCRIPTION
**Pull request by darquesdev**
**Motivation**
Currently, the content recommendation feature only works using the default company (virtual instance), this story was created to support different companies to the default company in Content Recommendation. 

**Approach**
I've exposed the companyId parameter as part of the AssetListAssetEntryQueryProcessor API, then I've updated the implementation to calculate the terms of an user depending on the current company.

**How to test it** 
I've added many unit tests, It's a bit tricky to create integration tests in this part. Functionally it is laborious to test it, I've added detailed steps as part of the [description of the story](https://issues.liferay.com/browse/LPS-111468).

### CODE REVIEW

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java [hunk 1]
```diff
@@ -22,6 +22,16 @@
  */
 public interface AssetListAssetEntryQueryProcessor {
 
+	public void processAssetEntryQuery(
```
Comment (cgoncas) [inline modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java:25]: Should @lifera-echo review this pull?

Comment (darquesdev) [reply modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java:25]: Good question! @dgarciasarai who was the reviewer of our first collab here?

**Comment by darquesdev** [comment]
ci:stop

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java:25]: This was the main _Story_ for Content Recommendation: https://issues.liferay.com/browse/LPS-87232.
I agree with you, I think @liferay-echo should review this.

Comment (darquesdev) [reply modules/apps/asset/asset-list-api/src/main/java/com/liferay/asset/list/asset/entry/query/processor/AssetListAssetEntryQueryProcessor.java:25]: I see It was by Edu in collaboration with @ealonso and @jorge.ferrer maybe It's good that some @liferay-echo dev takes a look ;)

**Comment by darquesdev** [comment]
After conversation with @cgoncas let's try to add an integration test ;)

**Comment by darquesdev** [comment]
Partially sent here https://github.com/liferay-echo/liferay-portal/pull/1817

**Pull request closed without merging**